src/SimplePie.php

```diff
@@ -1924,6 +1924,7 @@ protected function fetch_data(&$cache)
                             $this->status_code = 0;
 
                             if ($this->force_cache_fallback) {
+                                $this->data['cache_expiration_time'] = $this->cache_duration + time();
```

Comment:
Also probably want to do the same for 304 response.

Reply:
Oh, I was wondering why does not the in-cache TTL remove the entry when it expires, which would break the 304 and it does, as you found out in https://github.com/FreshRSS/simplepie/blob/5e92ce767597aa9f5cbb00d88c7591b2e604dc5f/src/Cache/BaseDataCache.php#L58-L62

Reply:
I can happily send upstream more of the fixes / improvements, but best after some of the existing pending PRs are processed